### PR TITLE
Fixed some server stuffs

### PR DESCRIPTION
**Pull request by hctankred**
Probably still need to modify Async stuff

### CODE REVIEW

File: src/interfaces/GameThread.java [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package interfaces;
+
+public interface GameThread {
+    public void send(String message);
```
Comment (jhviggo) [inline src/interfaces/GameThread.java:4]: There is no javaDoc :crying_cat_face:

Comment (hctankred) [reply src/interfaces/GameThread.java:4]: 🔪 😿

File: src/server/GameServer.java [hunk 2]
```diff
@@ -1,12 +1,17 @@
...
+import java.io.DataOutputStream;
 import java.io.IOException;
 import java.net.ServerSocket;
 import java.net.Socket;
+import java.util.ArrayList;
 
 public class GameServer {
     private static final int PORT = 1337;
     private static boolean SERVER_RUNNING = true;
+    private static ArrayList<GameServerThread> connections = new ArrayList<>();
```
Comment (jhviggo) [inline src/server/GameServer.java]: You could make this a set

File: src/interfaces/GameThread.java (hunk 1 above)
Comment (jhviggo) [reply src/interfaces/GameThread.java:4]: REEEE :poland:

File: src/server/GameServerThread.java [hunk 3]
```diff
@@ -17,11 +21,18 @@ public GameServerThread(final Socket connectionSocket) {
 
     public void run() {
         try {
-            final BufferedReader inFromClient = new BufferedReader(new InputStreamReader(connectionSocket.getInputStream()));
-            final DataOutputStream outToClient = new DataOutputStream(connectionSocket.getOutputStream());
-
-            outToClient.writeBytes("Sir lancelot\n");
-
+            AsyncReceive reader = new AsyncReceive(
```
Comment (jhviggo) [inline src/server/GameServerThread.java]: Why is this called reader? It is not a reader!

**Review by jhviggo — APPROVED** [review]

**Pull request merged**